lms/djangoapps/edraak_university/views.py

```diff
@@ -157,6 +160,8 @@ def is_form_disabled(self):
     @method_decorator(login_required)
     @method_decorator(ensure_valid_course_key)
     def dispatch(self, *args, **kwargs):
+        if not self.get_course().enable_university_id:
+            raise Http404
```

Comment:
It is not the best practice to suddenly disable a link that was active moments before, but rather redirect user to course root